### DISCUSSION TITLE
Which sections would you like to have in this project?

### DISCUSSION THREAD

**Discussion by FrancescoXX** [FrancescoXX/rustcrab#22] [Announcements]
Let's discuss the sections of this project.

Ideas:
- course
- books
- projects
- dev tools
- people

**Comment by FrancescoXX** [comment]
for now we have:
- books
- projects
- dev tools

coming soon: people and dsa

**Reply by Shivam-dev925** [reply]
Repository should have unit tests also

**Reply by FrancescoXX** [reply]
This comment is not relevant to the question, but let me elaborate

I would keep this optional for now, as I want this project to be beginner-friendly. but if you want to add tests for a section, just open an issue.

**Comment by felixoder** [comment]
May be we should work with "Learn Rust" where we will share topic wise discussions over rust may be some sidebar having the contents like 
Loops : some descriptions and some code snippets. and I think people can include some topic future to make it engaging. I hope you understood what I wanna say:. 
![image](https://github.com/user-attachments/assets/abd3583d-09c6-4694-b2dc-2997d6fd4316)
You can take ref like that.

**Reply by felixoder** [reply]
and there is no "this types" references for Rust if you do that it would be awesome

**Comment by FrancescoXX** [comment]
this website is mor emeant to bee as a recap of all the things, but we might integrate something about the basic concepts

**Reply by felixoder** [reply]
can I work over that may be we would add some basic concepts like intro,loop,enum,iteration,hm, generics etc. If you like that I can work with that

**Reply by FrancescoXX** [reply]
It's an idea. We can create a section called fundamentals, give a short explanation and then I can link my videos (I am creating all of them, basically)

you can refer to this playlist

https://www.youtube.com/watch?v=R33h77nrMqc&list=PLPoSdR46FgI412aItyJhj2bF66cudB6Qs

please open an issue

**Comment by anshumalivfx** [comment]
How about we add an online Compiler/Playground aswell for beginners?
I could start working on that if you approve this idea

**Reply by FrancescoXX** [reply]
do you mean a playground on the website? that would be so useful!

**Reply by anshumalivfx** [reply]
Yesss To let the user run Rust code without actually installing compiler on their machines 
It would be great option or beginners

**Reply by FrancescoXX** [reply]
it seems a great idea. We can create an issue to "create a compiler section" just to have something to start wtih

I would definitively use this in some videos

**Reply by anshumalivfx** [reply]
Sure I'll Create this Issue

**Reply by Alidantech** [reply]
How do playgrounds actually work, I have always asked myself. This will be a good opportunity to find out.

**Reply by anshumalivfx** [reply]
Well The Playgrounds work with REST API in this case

**Reply by Alidantech** [reply]
so we will need a server with rust installed then we will be sending code via api for it to be run.

**Reply by anshumalivfx** [reply]
We are already using 3rd party server but we want to create our own

**Comment by Alidantech** [comment]
I can work on the course section, to offer a more guided approach to learners

**Reply by anshumalivfx** [reply]
That'll be great you could show case your work here before making a pull request

**Reply by Alidantech** [reply]
okay then, understood
